kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/BaseOperation.java

```diff
@@ -222,7 +223,9 @@ public T getMandatory() {
     if (item != null) {
       return item;
     }
-    throw new KubernetesClientException("Resource not found!", 404, new StatusBuilder().withCode(404).withMessage("Not found.").build());
```

Comment:
Not sure what you mean by better status.

If you are talking about the status message that is part of KubernetesClientException, then yes it does have a nicer message now.